src/aircode.ts

```diff
@@ -304,21 +355,28 @@ export class AirCode implements vscode.FileSystemProvider {
             }
         };
 
-        ws.onclose = function (event: CloseEvent) {
+        ws.onclose = async function (event: CloseEvent) {
+            airCode.closingSocket = true;
+
             for (let [id, promise] of parent.promises) {
                 promise({
                     error: "connectionLost"
                 });
             }
             parent.promises.clear();
             parent.projectName = undefined;
-            vscode.window.showErrorMessage(`Connection lost to ${host}.`);
+
+            statusBarResource?.dispose();
+            statusBarResource = vscode.window.setStatusBarMessage(`Connection lost to ${host}`);
             if (event.code != CloseEventCode.IncompatibleVersion) {
-                vscode.window.showErrorMessage(`Connection lost to ${host}.`);
+                statusBarResource?.dispose();
+                statusBarResource = vscode.window.setStatusBarMessage(`Connection lost to ${host}`);
```

Comment:
Do we need this twice? I wasn't sure and simply replaced all instances of the previous messages. Seems like this one may be unnecessary? Was going to ask you

Reply:
Oh, actually we can probably get rid of the check there and the duplicate dispose/message.